.github/workflows/finalize_release.yml

```diff
@@ -0,0 +1,63 @@
+name: Finalize Release
+
+# Every time we trigger the workflow on a branch.
+on:
+  workflow_dispatch:
+
+jobs:
+  generate_version_name:
+    runs-on: ubuntu-latest
+
+    outputs:
+      version_name: ${{ steps.version_name.outputs.PROJECT_VERSION }}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      # Get the version name from a script and save it in version_name output.
+      - name: Generate version_name
+        id: version_name
+        run: |
+          echo "▸ Set run permission."
+          chmod +x scripts/version_name.sh
+          echo "▸ Getting version name"
+          PROJECT_VERSION=$(./scripts/version_name.sh)
+          echo "▸ Creating the version_name output with value: $PROJECT_VERSION"
+          echo "PROJECT_VERSION=$PROJECT_VERSION" >> $GITHUB_OUTPUT
+          echo "▸ DONE"
+
+  publish_draft_release:
+    runs-on: ubuntu-latest
+    needs: generate_version_name
+
+    steps:
+      - uses: actions/checkout@v4
+
+      # Publish Github release and the tag
+      - name: Publish Github release
+        uses: ncipollo/release-action@v1
+        env:
+          VERSION_NAME: ${{ needs.generate_version_name.outputs.version_name }}
+        with:
+          tag: ${{ env.VERSION_NAME }}
+          allowUpdates: true
```

Comment:
What does this actually mean? Would publishing the same version name override the previous release?

Reply:
Indeed. If `allowUpdates` is true, then the release gets updated if it already exists, based on `tag`. I also set the `updateOnlyUnreleased` flag to true, which will only update a release if it is not published yet.

Reply:
Ahh I see, that makes sense